package/MDAnalysis/analysis/pca.py

```diff
@@ -0,0 +1,270 @@
+                                                     self._ref_atom_positions,
+                                                     self._atoms,
+                                                     mobile_com=mobile_cog,
+                                                     ref_com=self._ref_cog)
+                    self.mean += mobile_atoms.positions.ravel()
+                else:
+                    self.mean += self._atoms.positions.ravel()
+
+            self.mean /= self.n_frames
+
+        # TODO mean structure should be accessible as an atomgroup object
+
+    def _single_frame(self):
+        if self.align:
+            mobile_cog = self._atoms.center_of_geometry()
+            mobile_atoms, old_rmsd = _fit_to(self._atoms.positions,
+                                             self._ref_atom_positions,
+                                             self._atoms,
+                                             mobile_com=mobile_cog,
+                                             ref_com=self._ref_cog)
+            # now all structures are aligned to reference
+            x = mobile_atoms.positions.ravel()
+        else:
+            x = self._atoms.positions.ravel()
+
+        x -= self.mean
+        self.cov += np.dot(x[:, np.newaxis], x[:, np.newaxis].T)
+
+    def _conclude(self):
+        self.cov /= self.n_frames - 1
+        e_vals, e_vects = np.linalg.eig(self.cov)
+        sort_idx = np.argsort(e_vals)[::-1]
+        self.variance = e_vals[sort_idx]
+        self.p_components = e_vects[:, sort_idx]
+        self.cumulated_variance = (np.cumsum(self.variance) /
+                                   np.sum(self.variance))
+        self._calculated = True
+
+    def transform(self, atomgroup, n_components=None, start=None, stop=None,
+                  step=None):
+        """Apply the dimensionality reduction on a trajectory
+
+        Parameters
+        ----------
+        atomgroup: MDAnalysis atomgroup
+            The atoms to be PCA transformed.
+        n_components: int, optional
+            The number of components to be projected onto, Default none: maps
+            onto all components.
+        start: int, optional
+            The frame to start on for the PCA transform. Default: None becomes
+            0, the first frame index.
+        stop: int, optional
+            Frame index to stop PCA transform. Default: None becomes n_frames.
+            Iteration stops *before* this frame number, which means that the
+            trajectory would be read until the end.
+        step: int, optional
+            Number of frames to skip over for PCA transform. Default: None
+            becomes 1.
+        Returns
+        -------
+        pca_space : array, shape (number of frames, number of components)
+        """
+        if not self._calculated:
+            self.run()
+        # TODO has to accept universe or trajectory slice here
+        if self._atoms.atoms != atomgroup.atoms:
```

Comment:
this will only check that the atomgroup has the same reference. That is to tight as a test. It should check that number and type of atoms matches.